File: bioskopina_mobile/lib/models/post.dart
import 'package:json_annotation/json_annotation.dart';
import '../models/comment.dart';
part 'post.g.dart';

@JsonSerializable()
class Post {
  int? id;
  int? userId;
  String? content;
  int? likesCount;
  int? dislikesCount;
  DateTime? datePosted;
  List<Comment>? comments;

  Post(
    this.id,
    this.userId,
    this.content,
    this.likesCount,
    this.dislikesCount,
    this.datePosted,
    this.comments,
  );

  factory Post.fromJson(Map<String, dynamic> json) => _$PostFromJson(json);

  Map<String, dynamic> toJson() => _$PostToJson(this);
}

File: backend/Bioskopina.Services/Database/Post.cs
using System;
using System.Collections.Generic;

namespace Bioskopina.Services.Database;

public partial class Post
{
    public int Id { get; set; }


    public int UserId { get; set; }

    public string Content { get; set; } = null!;

    public int LikesCount { get; set; }

    public int DislikesCount { get; set; }

    public DateTime DatePosted { get; set; }

    public string? VideoUrl { get; set; }


    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public virtual User User { get; set; } = null!;

    public virtual ICollection<UserPostAction> UserPostActions { get; set; } = new List<UserPostAction>();
  
}
